src/tauon/t_modules/t_main.py

```
@@ -7642,6 +7675,84 @@ def clear_lyrics_deco(self, track_object: TrackClass) -> list[ColourRGBA | None]
+				if find_synced_lyric_data(track_object, just_check=True):
+					lyrics_file.write( _("NOTE: You are editing synced lyrics. Please don't break the formatting, because we can't correct for that yet. Thanks") + '\n')
+				lyrics_file.write( track_object.lyrics )
+			else:
+				lyrics_file.write( _("Put the lyrics in this file."))
+		if self.system == "Windows" or self.msys:
+			os.startfile(target)
+		elif self.macos:
+			subprocess.call(["open", "-t", target])
+		else:
+			subprocess.call(["xdg-open", target])
+		self.gui.opened_lyric_file = True
+		self.show_message(_("Lyrics file opened."), _('Click "Reload" if you made any changes'), mode="arrow")
+
+
+	def write_lyrics(self, track: TrackClass) -> None:
+		if track.file_ext == "MP3":
+			audio = mutagen.id3.ID3(track.fullpath)
+			audio.add( mutagen.id3.USLT( text=track.lyrics ) )
+			audio.save()
+			logging.info(f"Edited lyrics in the file for {track.artist} - {track.title}")
+		elif track.file_ext == "FLAC":
+			audio = mutagen.flac.FLAC(track.fullpath)
+			audio["LYRICS"] = track.lyrics
+			audio.save()
+			logging.info(f"Edited lyrics in the file for {track.artist} - {track.title}")
+		elif track.file_ext in ("OPUS", "OGG"):
+			audio = mutagen.oggvorbis.OggVorbis(track.fullpath)
+			audio["LYRICS"] = track.lyrics
+			audio.save()
+			logging.info(f"Edited lyrics in the file for {track.artist} - {track.title}")
+		elif track.file_ext in ("APE","WV","TTA"):
+			audio = mutagen.apev2.APEv2(track.fullpath)
+			audio["Lyrics"] = track.lyrics
+			audio.save()
+			logging.info(f"Edited lyrics in the file for {track.artist} - {track.title}")
+		elif track.file_ext in ("MP4","M4A","M4B","M4P"):
+			audio = mutagen.mp4.MP4(track.fullpath)
+			audio['\xa9lyr'] = track.lyrics
+			audio.save()
+			logging.info(f"Edited lyrics in the file for {track.artist} - {track.title}")
+		else:
+			logging.info(f"Could not save lyrics to file with extension {track.file_ext}")
```

We need a GUI facing error message for this failure. I don't think we should fail silently.

---

fair